.dotnet/src/Custom/Chat/AssistantChatMessage.Serialization.cs

```diff
@@ -17,47 +17,13 @@ internal static void SerializeAssistantChatMessage(AssistantChatMessage instance
     protected override void WriteCore(Utf8JsonWriter writer, ModelReaderWriterOptions options)
     {
         writer.WriteStartObject();
-        if (Optional.IsDefined(ParticipantName))
-        {
-            writer.WritePropertyName("name"u8);
-            writer.WriteStringValue(ParticipantName);
-        }
-        if (Optional.IsCollectionDefined(ToolCalls))
-        {
-            writer.WritePropertyName("tool_calls"u8);
-            writer.WriteStartArray();
-            foreach (var item in ToolCalls)
-            {
-                writer.WriteObjectValue(item, options);
-            }
-            writer.WriteEndArray();
-        }
-        if (Optional.IsDefined(FunctionCall))
-        {
-            if (FunctionCall != null)
-            {
-                writer.WritePropertyName("function_call"u8);
-                writer.WriteObjectValue(FunctionCall, options);
-            }
-            else
-            {
-                writer.WriteNull("function_call");
-            }
-        }
         writer.WritePropertyName("role"u8);
-        writer.WriteStringValue(Role);
-        if (Optional.IsCollectionDefined(Content))
-        {
-            if (Content[0] != null)
-            {
-                writer.WritePropertyName("content"u8);
-                writer.WriteStringValue(Content[0].Text);
-            }
-            else
-            {
-                writer.WriteNull("content");
-            }
-        }
+        writer.WriteStringValue(Role.ToSerialString());
+        ChatMessageContentPart.WriteCoreContentPartList(Content, writer, options);
+        writer.WriteOptionalProperty("refusal"u8, Refusal, options);
+        writer.WriteOptionalProperty("name"u8, ParticipantName, options);
+        writer.WriteOptionalCollection("tool_calls"u8, ToolCalls, options);
+        writer.WriteOptionalProperty("function_call"u8, FunctionCall, options);
```

Comment:
They never had explicit nullability; part of the goal of refactoring the repeated patterns into these helpers was to make that more clear than it otherwise is.

This was the previous logic for `function_call`:

```csharp
        if (Optional.IsDefined(FunctionCall))
        {
            if (FunctionCall != null)
            {
                writer.WritePropertyName("function_call"u8);
                writer.WriteObjectValue(FunctionCall, options);
            }
            else
            {
                writer.WriteNull("function_call");
            }
        }
```

The implementation of `Optional.IsDefined` is:

```csharp
        public static bool IsDefined(object value)
        {
            return value != null;
        }
```

Thus, we've never had the ability to explicitly put null onto the wire via convenience methods and the generator doesn't really allow that; nullability just influences whether we'll throw an exception for a null check.